consul/autopilot.go

```diff
@@ -221,39 +220,87 @@ func (s *Server) serverHealthLoop() {
 				break
 			}
 
-			// Build an updated map of server healths
+			// Get the the serf members which are Consul servers
+			serverMap := make(map[string]serf.Member)
 			for _, member := range s.LANMembers() {
 				if member.Status == serf.StatusLeft {
 					continue
 				}
 
 				valid, parts := agent.IsConsulServer(member)
 				if valid {
-					health, err := s.queryServerHealth(member, parts, autopilotConf)
-					if err != nil {
-						s.logger.Printf("[ERR] consul: error fetching server health: %s", err)
-						serverHealths[parts.ID] = &structs.ServerHealth{
-							ID:      parts.ID,
-							Name:    parts.Name,
-							Healthy: false,
-						}
-					} else {
-						serverHealths[parts.ID] = health
-					}
+					serverMap[parts.ID] = member
+				}
+			}
+
+			future := s.raft.GetConfiguration()
+			if err := future.Error(); err != nil {
+				s.logger.Printf("[ERR] consul: error getting Raft configuration %s", err)
+				break
+			}
+
+			// Build a current list of server healths
+			var clusterHealth structs.OperatorHealthReply
+			servers := future.Configuration().Servers
+			healthyCount := 0
+			voterCount := 0
+			for _, server := range servers {
+				member, ok := serverMap[string(server.ID)]
```

Comment:
It would be cleaner to have a health record for each thing in the Raft configuration. It seems like we could do something like:

```
health := &structs.ServerHealth{
    ID: server.ID,
}

member, ok := serverMap[string(server.ID)]
if ok {
    health.SerfStatus = member.Status
} else {
    health.SerfStatus = "unknown"
}

// fill in the Raft parts
// ...
clusterHealth.Servers = append(clusterHealth.Servers, *health)
```

I think `queryServerHealth()` could be modified to just take the server ID. We should change this to be ID-based:

```
        // Set LastContact to 0 for the leader
        if s.config.NodeName == member.Name {
                health.LastContact = 0
        }
```